### PR TITLE
Handle local prefixes special when merging linkml files

### PR DESCRIPTION
**Pull request by jankatins**
This can e.g. happen if we have this kind of filestructure:

```
lib.yaml
folder/subfolder/more_stuff.yaml -> references as ../../lib.yaml
folder/all.yaml -> references both ../lib.yaml and ./subfolder/more_stuff.yaml
```

Now we want to use `gen-yaml` on `folder/all.yml` to create a combined linkml file

### CODE REVIEW

**Comment by cmungall** [comment]
it would be great to have a unit test for this - is this something you'd be able to take on or shall we have some of the core team handle it?

**Comment by jankatins** [comment]
huih. My only idea would be to write out some very basic linkml structure with three files and then run yamlgen on it. I tried to find some tests for `merge_namespaces()` but couldn't find any.

Would that work for you?

**Comment by cmungall** [comment]
that would be ideal, thanks!

File: linkml/utils/mergeutils.py [hunk 1]
```diff
@@ -73,6 +74,37 @@ def merge_namespaces(
...
+                    and target.prefixes[prefix.prefix_prefix].prefix_reference
+                    != prefix.prefix_reference
+                ):
+                    logging.info(
+                        "Ignoring different relative prefix for %s from %s, "
+                        + "as we cannot adjust it relative to the final file. "
+                        + "Assuming the first found location is correct: %s!",
+                        prefix.prefix_prefix,
+                        mergee.name,
+                        namespaces[prefix.prefix_prefix],
```
Comment (sujaypatil96) [inline linkml/utils/mergeutils.py:104]: A good place to use Python [f-strings](https://peps.python.org/pep-0498/)?

File: linkml/utils/mergeutils.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import dataclasses
+import logging
```
Comment (sujaypatil96) [inline linkml/utils/mergeutils.py:2]: Probably a good idea to follow syntax like this: https://github.com/linkml/linkml/blob/main/linkml/generators/linkmlgen.py#L14?

**Review by sujaypatil96 — COMMENTED** [review]
Left a couple of very trivial style guide review comments. Great work on the PR @jankatins

File: linkml/utils/mergeutils.py (hunk 1 above)
Comment (cmungall) [reply linkml/utils/mergeutils.py:104]: yes, see also: https://stackoverflow.com/questions/45965007/multiline-f-string-in-python

but we can do this after merging

Comment (jankatins) [reply linkml/utils/mergeutils.py:104]: the linters won't like it: no fstring in the `log.*()` functions, I think for performance reasons (I actually converted it to use log and had a print(f"...") here in my q&d code :)

**Pull request merged**